x-pack/legacy/plugins/siem/public/containers/hosts/filter.tsx

```diff
@@ -51,53 +60,89 @@ interface HostsFilterDispatchProps {
 
 export type HostsFilterProps = OwnProps & HostsFilterReduxProps & HostsFilterDispatchProps;
 
-const HostsFilterComponent = pure<HostsFilterProps>(
-  ({
-    applyHostsFilterQuery,
-    children,
-    hostsFilterQueryDraft,
-    indexPattern,
-    isHostFilterQueryDraftValid,
-    setHostsFilterQueryDraft,
-    type,
-  }) => (
-    <>
-      {children({
-        applyFilterQueryFromKueryExpression: (expression: string) =>
-          applyHostsFilterQuery({
-            filterQuery: {
-              kuery: {
-                kind: 'kuery',
-                expression,
-              },
-              serializedQuery: convertKueryToElasticSearchQuery(expression, indexPattern),
-            },
-            hostsType: type,
-          }),
-        filterQueryDraft: hostsFilterQueryDraft,
-        isFilterQueryDraftValid: isHostFilterQueryDraftValid,
-        setFilterQueryDraftFromKueryExpression: (expression: string) =>
-          setHostsFilterQueryDraft({
-            filterQueryDraft: {
-              kind: 'kuery',
-              expression,
-            },
-            hostsType: type,
-          }),
-      })}
-    </>
-  )
-);
+class HostsFilterComponent extends React.PureComponent<HostsFilterProps> {
+  private memoizedApplyFilterQueryFromKueryExpression: (expression: string) => void;
+  private memoizedSetFilterQueryDraftFromKueryExpression: (expression: string) => void;
 
-HostsFilterComponent.displayName = 'HostsFilterComponent';
+  constructor(props: HostsFilterProps) {
+    super(props);
+    this.memoizedApplyFilterQueryFromKueryExpression = memoizeOne(
+      this.applyFilterQueryFromKueryExpression
+    );
+    this.memoizedSetFilterQueryDraftFromKueryExpression = memoizeOne(
+      this.setFilterQueryDraftFromKueryExpression
+    );
+  }
+
+  public render() {
+    const {
+      children,
+      hostsFilterQueryDraft,
+      indexPattern,
+      kueryFilterQuery,
+      isHostFilterQueryDraftValid,
+      setQuery,
+      type,
+    } = this.props;
+
+    if (setQuery) {
+      setQuery({
```

Comment:
Given that `setQuery` is an `actionCreator`:

```
export const setQuery = actionCreator<{
  inputId: InputsModelId;
  id: string;
  loading: boolean;
  refetch: Refetch;
  inspect: InspectQuery | null;
}>('SET_QUERY');
```

dispatching `setQuery` inside `render()` here feels like it violates:

> The render() function should be pure, meaning that it does not modify component state, it returns the same result each time it’s invoked, and it does not directly interact with the browser.

as documented here https://reactjs.org/docs/react-component.html#render

, because dispatching an action creates an effect (a state change).